pkg/config/config_test.go

```diff
@@ -0,0 +1,130 @@
+	. "github.com/onsi/gomega"
+	"github.com/projectcalico/k8s-policy/pkg/config"
+	"os"
+)
+
+var _ = Describe("Config", func() {
+
+	// unsetEnv() function that unsets environment variables
+	// required by k8s-policy controller
+	unsetEnv := func() {
+		os.Unsetenv("LOG_LEVEL")
+		os.Unsetenv("RECONCILER_PERIOD")
+		os.Unsetenv("ENABLED_CONTROLLERS")
+		os.Unsetenv("ENDPOINT_WORKERS")
+		os.Unsetenv("PROFILE_WORKERS")
+		os.Unsetenv("POLICY_WORKERS")
+		os.Unsetenv("KUBECONFIG")
+	}
+
+	// setEnv() function that sets environment variables
+	// to some sensbile values
+	setEnv := func() {
+		os.Setenv("LOG_LEVEL", "debug")
+		os.Setenv("RECONCILER_PERIOD", "2m5s")
+		os.Setenv("ENABLED_CONTROLLERS", "policy")
+		os.Setenv("ENDPOINT_WORKERS", "1")
+		os.Setenv("PROFILE_WORKERS", "3")
+		os.Setenv("POLICY_WORKERS", "3")
+		os.Setenv("KUBECONFIG", "/home/user/.kube/config")
+	}
+
+	// setWrongEnv() function sets environment variables
+	// with values of wrong data type
+	setWrongEnv := func() {
+		os.Setenv("ENDPOINT_WORKERS", "somestring")
+		os.Setenv("PROFILE_WORKERS", "somestring")
+		os.Setenv("POLICY_WORKERS", "somestring")
+	}
+
+	Context("with default values", func() {
+
+		// Unset environment variables
+		unsetEnv()
+
+		// Parse config
+		config := new(config.Config)
+		err := config.Parse()
+
+		// Assert no error generated
+		It("shoud not generate error", func() {
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		// Assert default values
+		It("shoud return default values", func() {
+			Expect(config.LogLevel).To(Equal("info"))
+			Expect(config.ReconcilerPeriod).To(Equal("5m"))
+			Expect(config.EnabledControllers).To(Equal("policy,profile,endpoint"))
+			Expect(config.EndpointWorkers).To(Equal(3))
+			Expect(config.ProfileWorkers).To(Equal(1))
+			Expect(config.PolicyWorkers).To(Equal(1))
+			Expect(config.Kubeconfig).To(Equal(""))
+		})
+	})
+
+	Context("with valid user defined values", func() {
+
+		// Set environment variables
+		setEnv()
+
+		// Reset environment variables
+		defer unsetEnv()
+
+		// Parse config
+		config := new(config.Config)
+		err := config.Parse()
+
+		// Assert no error generated
+		It("shoud not generate error", func() {
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		// Assert values
+		It("shoud return user defined values", func() {
+			Expect(config.LogLevel).To(Equal("debug"))
+			Expect(config.ReconcilerPeriod).To(Equal("2m5s"))
+			Expect(config.EnabledControllers).To(Equal("policy"))
+			Expect(config.EndpointWorkers).To(Equal(1))
+			Expect(config.ProfileWorkers).To(Equal(3))
+			Expect(config.PolicyWorkers).To(Equal(3))
+			Expect(config.Kubeconfig).To(Equal("/home/user/.kube/config"))
+		})
+	})
+
+	Context("with invalid user defined values", func() {
+
+		// Set wrong environment variables
+		setWrongEnv()
+
+		// Reset environment variables
+		defer unsetEnv()
+
+		// Parse config
+		config := new(config.Config)
+		err := config.Parse()
+
+		// Assert error is generated
```

Comment:
This comment is wrong.

Reply:
@caseydavenport, fixed the spec description that it should generate an error.